### PR TITLE
Use compute-clusters in config

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
-  Support `compute-clusters` in config instead of top-level mesos parameters

## Why are we making these changes?
Gets us ready to support kubernetes

### CODE REVIEW

File: scheduler/src/cook/config.clj [hunk 1]
```diff
@@ -335,11 +343,13 @@
      :mesos-framework-name (fnk [[:config {mesos nil}]]
                              (when mesos
                                (or (:framework-name mesos) "Cook")))
-     :mesos-framework-id (fnk [[:config {mesos nil}]]
-                           (:framework-id mesos))
-     :mesos-compute-cluster-name (fnk [[:config {mesos nil}]]
-                                   (or (:compute-cluster-name mesos) "default-compute-cluster-from-config-defaulting"))
-
+     ; TODO(pschorf): Remove
```
Comment (dposada) [inline scheduler/src/cook/config.clj]: Is this TODO for this PR or a future one?

File: scheduler/src/cook/mesos/mesos_compute_cluster.clj [hunk 2]
```diff
@@ -250,77 +250,48 @@
...
+  "Checks datomic for a compute cluster with the given name and framework-id. If present, returns the entity id.
+   If missing, installs in datomic and returns the entity id."
+  [conn compute-cluster-name framework-id]
+  (let [compute-cluster-entity-id (get-mesos-cluster-entity-id (d/db conn)
+                                                               {:compute-cluster-name compute-cluster-name
+                                                                :framework-id framework-id})]
+    (if compute-cluster-entity-id
+      compute-cluster-entity-id
+      (cc/write-compute-cluster conn (mesos-cluster->compute-cluster-map-for-datomic {:compute-cluster-name compute-cluster-name
+                                                                                      :framework-id framework-id})))))
```
Comment (dposada) [inline scheduler/src/cook/mesos/mesos_compute_cluster.clj]: Missing newline here

File: integration/travis/scheduler_travis_config.edn [hunk 3]
```diff
@@ -61,8 +64,7 @@
  :mesos {:master #config/env "MINIMESOS_ZOOKEEPER"
          :failover-timeout-ms nil
          :leader-path "/cook-scheduler"
-         :role "cook"
-         :framework-id #config/env "COOK_FRAMEWORK_ID"}
+         :role "cook"}
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/travis/scheduler_travis_config.edn]: All but two of these can be moved: the other two are global. I added a TODO to rename them to keep the scope here down.

File: scheduler/src/cook/config.clj (hunk 1 above)
Comment (pschorf) [reply scheduler/src/cook/config.clj]: Future PR (currently WIP, but still want to merge separately)

File: scheduler/src/cook/mesos/mesos_compute_cluster.clj (hunk 2 above)
Comment (pschorf) [reply scheduler/src/cook/mesos/mesos_compute_cluster.clj]: Done

**Pull request merged**